generate-err-logs.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash -e
+
+
+main() {
+    echo "Generating error logs" >&2
+    generate_logs
+    echo "Generated logs sucessfully" >&2
+}
+
+generate_logs() {
+    pod_definitions_file="failed-pods-definitions.yaml"
+    pod_logs_file="failed-pods-logs.log"
+    event_messages_file="failed-deployment-event-log.log"
+    > $pod_logs_file > $pod_definitions_file > $event_messages_file
+
+    namespaces=$(kubectl get namespaces -o name | xargs -n1 basename)
+
+    for namespace in $namespaces; do
+        events=$(kubectl get events -n $namespace --field-selector type=Warning -o jsonpath='{range .items[*]}{.involvedObject.kind}{" "}{.involvedObject.name}{" "}{.message}{" ("}{.reason}{")\n"}{end}')
```

Comment:
Can you add a comment for what this line does?

Reply:
Added. also broke the lines to add more clarity.